fun main() {
  let a = 0;

  try {
    a = 3;
    throw "abc";
  } catch y: IntArray {
    println("I should not be reached.");
    a = 4;

  } catch x: Str {
    a = 5;
  }

  assert(a == 5);
}
